metrics/core/metrics.go

```diff
@@ -141,6 +141,14 @@ var NetworkMetrics = []Metric{
 	MetricNetworkTxRate,
 }
 
+// Maps from non primary resource name (i.e., not cpu or memory) to the metric
+// that tracks container resource request on that resource. The name of the metric
+// is in format of formattedResourceName/request where formattedExtendedResourceName
+// is generated from the resource name with "/" replaced by "_".
+// E.g., a container requesting nvidia.com/gpu resource will generates a
+// nvidia.com_gpu/request metric in this map.
+var NonPrimaryResourceRequestMetrics = map[string]Metric{}
```

Comment:
Can you put all resource request metrics here, both primary and extended? You can initialize a map with existing primary resource metric. I don't think the logic should be different for extended resources.

What about limits?